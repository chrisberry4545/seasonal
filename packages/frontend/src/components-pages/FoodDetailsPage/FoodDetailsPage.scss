@import './../../../node_modules/@chrisb-dev/seasonal-shared-frontend-components/dist/styles/settings';
@import './../../../node_modules/@chrisb-dev/seasonal-shared-frontend-components/dist/styles/tools';

  .c-food-details-page__main {
    max-width: 790px;
    margin: auto;
    width: 100%;
    padding: 0 2rem;
  }
