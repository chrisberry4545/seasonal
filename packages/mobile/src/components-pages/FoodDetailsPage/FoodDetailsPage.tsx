import React, { FC } from 'react';

import {
  FoodDetailsLoaderConnecter,
  FoodDetailsTopSectionConnecter,
  SeasonsForFoodConnecter,
  HeaderConnecter,
  RecipesForFoodConnecter
} from '../../components-main';

import { MainContainer, DefaultPaddingContainer } from '../../components-layout';
import { ScrollView } from 'react-native-gesture-handler';

export const FoodDetailsPage: FC<{}> = () => (
  <MainContainer>
    <HeaderConnecter />
    <DefaultPaddingContainer>
      <ScrollView>
        <FoodDetailsLoaderConnecter />
        <FoodDetailsTopSectionConnecter />
        <RecipesForFoodConnecter />
        <SeasonsForFoodConnecter />
      </ScrollView>
    </DefaultPaddingContainer>
  </MainContainer>
);
