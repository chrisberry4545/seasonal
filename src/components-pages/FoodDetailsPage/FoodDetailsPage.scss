@import '../../styles/settings';
@import '../../styles/tools';

.c-food-details-page {
  margin-top: $default-top-margin;
  @include mq($from: tablet) {
    display: flex;
  }
}

  .c-food-details-page__main {
    max-width: 750px;
    margin: auto;
    width: 100%;
  }
