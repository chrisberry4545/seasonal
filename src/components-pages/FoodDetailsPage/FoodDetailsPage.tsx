import React, { SFC } from 'react';
import { NavigationScreenProp } from 'react-navigation';

import {
  FoodDetailsLoaderConnecter,
  FoodDetailsTopSectionConnecter,
  SeasonsForFoodConnecter,
  NavigationBar
} from '../../components-main';

import { MainContainer } from '../../components-layout';

export const FoodDetailsPage: SFC<{
  navigation: NavigationScreenProp<{}>
}> = ({ navigation }) => (
  <MainContainer>
    <NavigationBar navigation={ navigation } />
    <FoodDetailsTopSectionConnecter navigation={ navigation } />
    <FoodDetailsLoaderConnecter />
    <SeasonsForFoodConnecter navigation={ navigation }/>
  </MainContainer>
);
