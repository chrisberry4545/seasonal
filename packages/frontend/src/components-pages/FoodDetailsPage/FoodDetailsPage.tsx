import React, { FC } from 'react';

import './FoodDetailsPage.scss';

import {
  FoodDetailsLoaderConnecter,
  FoodDetailsTopSectionConnecter,
  PageWithMenu,
  SeasonsForFoodConnecter,
  RecipesForFoodConnecter,
  FoodBadgesConnecter
} from '../../components-main';

export const FoodDetailsPage: FC<{}> = () => (
  <PageWithMenu>
    <div className='c-food-details-page__main'>
      <FoodDetailsTopSectionConnecter />
      <FoodBadgesConnecter />
      <FoodDetailsLoaderConnecter />
      <RecipesForFoodConnecter />
      <SeasonsForFoodConnecter />
    </div>
  </PageWithMenu>
);
