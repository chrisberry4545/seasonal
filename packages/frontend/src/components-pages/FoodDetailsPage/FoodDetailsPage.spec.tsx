import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { FoodDetailsPage } from '../FoodDetailsPage/FoodDetailsPage';

jest.mock('../../components-main', () => ({
  FoodDetailsLoaderConnecter: () => 'FoodDetailsLoaderConnecter',
  FoodDetailsTopSectionConnecter: () => 'FoodDetailsTopSectionConnecter',
  PageWithMenu: () => 'PageWithMenu',
  RecipesForFoodConnecter: () => 'RecipesForFoodConnecter',
  SeasonsForFoodConnecter: () => 'SeasonsForFoodConnecter'
}));

describe('<FoodDetailsPage />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() =>
    wrapper = shallow(
      <FoodDetailsPage />
    )
  );

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

});
