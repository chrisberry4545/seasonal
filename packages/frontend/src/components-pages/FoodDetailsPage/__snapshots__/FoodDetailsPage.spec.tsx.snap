// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoodDetailsPage /> renders correctly 1`] = `
<PageWithMenu>
  <div
    className="c-food-details-page__main"
  >
    <FoodDetailsTopSectionConnecter />
    <FoodDetailsLoaderConnecter />
    <RecipesForFoodConnecter />
    <SeasonsForFoodConnecter />
  </div>
</PageWithMenu>
`;
